DM-44150: Remove explicit SdssShape dependency

parejkoj opened the pull request.

bsmartradio commented on tests/test_trailedSources.py:
I'm not sure why github isn't letting me select specific lines, but there are references to `SdssShape` on lines 188,226, 227,253, 318. Will these be left in until we refactor the tests?

parejkoj replied:
You can't comment on lines that aren't part of the PR, sadly.

With these changes, the trailed plugin is agnostic to which shape plug is used, so using SdssShape in the tests is fine. This also means that we don't have to add a dependency on meas_extensions_shapeHsm to this package, which is a good thing.

bsmartradio commented on tests/test_trailedEdgeSources.py:
There are references to `base_SdssShape` on lines 66, 169, 270, and 376 instead of `HSMShape` or `slot_shape`.

parejkoj replied:
As I noted above, that's fine: trailed sources is now agnostic to which shape plugin was run, so we can use whatever is easiest in the tests.

bsmartradio reviewed (changes requested):
The changes look fine, and the change in handling the length measurement is how I was going to change it. My only other comment is that there are a number of `SdssShape` references in both unit tests.

bsmartradio reviewed (approved):
As said above, `SdssShape` can be used for the unit tests, so everything looks good.

parejkoj commented:
@bsmartradio : please take a look at the new test I added for you to expand on. I put in docstrings and refactored it to be easy to add new tests to, but let me know if anything there doesn't make sense.

bsmartradio reviewed (approved):
The new addition to the unit tests looks good.
